cloud/ovirt/ovirt_quotas_facts.py

```
@@ -0,0 +1,112 @@
+    name:
+        description:
+            - "Name of the quota, can be used as glob expression."
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about quota named C<myquota> in Default datacenter:
+- ovirt_quotas_facts:
+    datacenter: Default
+    name: myquota
+- debug:
+    var: ovirt_quotas
+'''
+
+RETURN = '''
+ovirt_quotas:
+    description: "List of dictionaries describing the quotas. Quota attribues are mapped to dictionary keys,
+                  all quotas attributes can be found at following url: https://ovirt.example.com/ovirt-engine/api/model#types/quota."
+    returned: On success.
+    type: list
+'''
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        datacenter=dict(required=True),
+        name=dict(default=None),
+    )
+    module = AnsibleModule(argument_spec)
+    check_sdk(module)
+
+    try:
+        connection = create_connection(module.params.pop('auth'))
+        datacenters_service = connection.system_service().data_centers_service()
+        dc_name = module.params['datacenter']
+        dc = search_by_name(datacenters_service, dc_name)
+        if dc is None:
+            raise Exception("Datacenter '%s' was not found." % dc_name)
+
+        quotas_service = datacenters_service.service(dc.id).quotas_service()
+        if module.params['name']:
+            quotas = [
+                e for e in quotas_service.list()
+                if fnmatch.fnmatch(e.name, module.params['name'])
+            ]
+        else:
+            quotas = quotas_service.list()
+
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_quotas=[
+                    get_dict_of_struct(c) for c in quotas
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Same here, please surface the full exception.